tests/Doctrine/Tests/ORM/Functional/Ticket/GH7941Test.php

```diff
@@ -68,7 +70,7 @@ public function typesShouldBeConvertedForDQLFunctions(): void
         foreach ($query->getResult() as $i => $item) {
             $product = self::PRODUCTS[$i];
 
-            self::assertSame(ltrim($product['price'], '-'), $item['absolute']);
+            self::assertEquals(ltrim($product['price'], '-'), $item['absolute']);
```

Comment:
What was the reason to lower strictness for this assert?

Reply:
The type returned by the database driver is beyond our control. In PHP 8.1, various PDO drivers have become more type-aware and return integers and floats where we've received a string previously.